test/scripts/link-stdlib.test.js

```diff
@@ -30,5 +30,10 @@ contract('link-stdlib command', function() {
     editJson(packageFileName, p => { p.lib = true; });
     await linkStdlib({ stdlibNameVersion: 'mock-stdlib@1.1.0', packageFileName }).should.be.rejectedWith(/lib/);
   });
-  
+
+  it('should raise an error if requested version of stdlib does not match its package version', async function () {
+    editJson('test/mocks/mock-stdlib/zos.json', p => p.version = '1.2.0')
+    await linkStdlib({ stdlibNameVersion: 'mock-stdlib@1.1.0', packageFileName }).should.be.rejectedWith('Requested version 1.1.0 does not match stdlib package version 1.2.0')
+    editJson('test/mocks/mock-stdlib/zos.json', p => p.version = '1.1.0')
```

Comment:
Note that if this test fails, it will fail to restore the zos.json file back to its correct state, and may be inadvertently commited back (since it is not ignored). Consider adding another mock-stdlib, where its package.json version does not match its zos.json.